### ISSUE TITLE
Panic in tokio-runtime-worker thread

### ISSUE THREAD

**Issue by bladedoyle** [mimblewimble/grin#3398]
**Describe the bug**
Fresh sync node.  Wallet restored from seed.  Attempt to get wallet info fails due to thread panic in the node.

**To Reproduce**
I synced new node.  Restored wallet from seed.  ran grin-wallet scan.  grin-wallet info

**Relevant Information**
```
$ grin-wallet info
Password: 
20200721 16:12:58.555 ERROR grin_wallet_impls::node_clients::http - Error calling get_kernel: Request error: Cannot make request: connection closed before message completed

```

```
20200721 16:06:51.286 ERROR grin_util::logger - 
thread 'tokio-runtime-worker' panicked at 'prune list 1-indexed, 0 not valid pos': store/src/prune_list.rs:271   0: grin_util::logger::send_panic_to_log::{{closure}}
   1: std::panicking::rust_panic_with_hook
             at /rustc/49cae55760da0a43428eba73abcb659bb70cf2e4/src/libstd/panicking.rs:515
   2: std::panicking::begin_panic
   3: grin_store::prune_list::PruneList::is_pruned
   4: <grin_store::pmmr::PMMRBackend<T> as grin_core::core::pmmr::backend::Backend<T>>::get_data
   5: grin_chain::txhashset::txhashset::PMMRHandle<grin_core::core::block::BlockHeader>::get_header_hash_by_height
   6: grin_chain::chain::Chain::get_header_by_height
   7: grin_chain::chain::Chain::get_kernel_height
   8: grin_api::handlers::chain_api::KernelHandler::get_kernel_v2
   9: <grin_api::foreign::Foreign<B,P,V> as grin_api::foreign_rpc::ForeignRpc>::get_kernel
  10: <dyn grin_api::foreign_rpc::ForeignRpc as easy_jsonrpc_mw::Handler>::handle
  11: easy_jsonrpc_mw::handle_call
  12: easy_jsonrpc_mw::handle_parsed_request
  13: easy_jsonrpc_mw::Handler::handle_request
  14: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  15: <hyper::proto::h1::dispatch::Server<S,hyper::body::body::Body> as hyper::proto::h1::dispatch::Dispatch>::poll_msg
  16: hyper::proto::h1::dispatch::Dispatcher<D,Bs,I,T>::poll_catch
  17: <hyper::server::conn::upgrades::UpgradeableConnection<I,S,E> as core::future::future::Future>::poll
  18: <hyper::server::conn::spawn_all::NewSvcTask<I,N,S,E,W> as core::future::future::Future>::poll
  19: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
  20: tokio::runtime::task::harness::Harness<T,S>::poll
  21: std::thread::local::LocalKey<T>::with
  22: tokio::runtime::thread_pool::worker::Context::run_task
  23: tokio::runtime::thread_pool::worker::Context::run
  24: tokio::macros::scoped_tls::ScopedKey<T>::set
  25: tokio::runtime::thread_pool::worker::run
  26: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
  27: tokio::runtime::task::harness::Harness<T,S>::poll
  28: tokio::runtime::blocking::pool::Inner::run
  29: tokio::runtime::context::enter
  30: std::sys_common::backtrace::__rust_begin_short_backtrace
  31: core::ops::function::FnOnce::call_once{{vtable.shim}}
  32: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/49cae55760da0a43428eba73abcb659bb70cf2e4/src/liballoc/boxed.rs:1008
      <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/49cae55760da0a43428eba73abcb659bb70cf2e4/src/liballoc/boxed.rs:1008
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/49cae55760da0a43428eba73abcb659bb70cf2e4/src/libstd/sys/unix/thread.rs:87
  33: start_thread
  34: __clone
```

**Comment by quentinlesceller** [comment]
Are you using Grin v4.0.1?

**Comment by bladedoyle** [comment]
4.0.0

**Comment by bladedoyle** [comment]
Ah, yes, this seems to be resolved in v4.0.1
Thanks @quentinlesceller
